test/FunctionalTests/Microsoft.OData.Core.Tests/Json/DefaultStreamBasedJsonWriterFactoryTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DefaultStreamBasedJsonWriterFactoryTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.OData.Json;
+using Xunit;
+
+namespace Microsoft.OData.Tests.Json
+{
+    public sealed class DefaultStreamBasedJsonWriterFactoryTests
+    {
+
+        #region Argument validation
+
+        [Fact]
+        public void CreateJsonWriter_ThrowsIfStreamIsNull()
+        {
+            DefaultStreamBasedJsonWriterFactory factory = DefaultStreamBasedJsonWriterFactory.Default;
+
+            Assert.Throws<ArgumentNullException>("stream",() => factory.CreateJsonWriter(null, false, Encoding.UTF8));
+        }
+
+        [Fact]
+        public void CreateJsonWriter_ThrowsIfEncodingIsNull()
+        {
+            DefaultStreamBasedJsonWriterFactory factory = DefaultStreamBasedJsonWriterFactory.Default;
+            using MemoryStream stream = new MemoryStream();
+
+            Assert.Throws<ArgumentNullException>("encoding", () => factory.CreateJsonWriter(stream, false, null));
+        }
+
+        #endregion
+
+        public static IEnumerable<object[]> Encodings
+           => new object[][]
+           {
+                new object[] { Encoding.UTF8 },
+                new object[] { Encoding.Unicode },
+                new object[] { Encoding.UTF32 },
+                new object[] { Encoding.BigEndianUnicode },
+                new object[] { Encoding.ASCII }
+           };

```

Comment:
```suggestion
        public static IEnumerable<IEnumerable<object>> Encodings { get; } = new object[][]
           {
                new object[] { Encoding.UTF8 },
                new object[] { Encoding.Unicode },
                new object[] { Encoding.UTF32 },
                new object[] { Encoding.BigEndianUnicode },
                new object[] { Encoding.ASCII }
           };
```

Does this need to be a field for xunit or anything? It seems like we wouldn't want to allocate new arrays every time it's called, unless test modifies it for some reason, but we probably shouldn't allow that either.

Reply:
No this was my mistake. Maybe I wrongly assumed the compiler caches that. I don't remember. But should definitely not be allocating new arrays.